### PR TITLE
AEA-3022 Provide a clear error message when no endorsement code is provided

### PR DESCRIPTION
**Pull request by kris-szlapa**
## Summary
:ticket: Jira Reference: [AEA-3022](https://nhsd-jira.digital.nhs.uk/browse/AEA-3022)
**Provide a clear error message when no endorsement code is provided.**
We currently provide a generic "UNKNOWN ERROR" message when a dispense claim is missing an endorsement code. This results in API consumers spending more time than they need to on integrating with our API.
This should be an update to the error message when no endorsement code isn't provided - this should be from a standard validation error message to respond to this. 

- Routine Change
- :exclamation: Breaking Change
- :sparkles: New Feature
- :warning: Potential issues that might be caused by this change

### Details
Acceptance Criteria
**GIVEN** a claim for a prescription
**WHEN** the claim does not have an endorsement
**THEN** the error message provided must clearly state that the error is a result of an endorsement not being provided

## Reviews Required

- [x] Dev
- [x] Test
- [ ] Tech Author
- [ ] Product Owner

## Review Checklist

:information_source: This section is to be filled in by the **reviewer**.

- [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
- [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
- [ ] I have ensured the jira ticket has been updated with the github pull request link

### CODE REVIEW

File: packages/coordinator/src/services/validation/claim-validator.ts [hunk 1]
```diff
@@ -60,5 +61,12 @@ export function verifyClaim(
     return permissionErrors
   }
 
+  const medicationDispenseEndorsement = getMedicationDispenseEndorsement(claim)
+  if (!medicationDispenseEndorsement.length) {
```
Comment (originalphil) [inline packages/coordinator/src/services/validation/claim-validator.ts:65]: This logic could be included in the getMedicationDispenseEndorsement function. Then you could call it medicationDispenseEndorsementPresent and return a boolean.

File: packages/coordinator/tests/services/validation/claim-validator.spec.ts [hunk 2]
```diff
@@ -21,4 +22,9 @@ describe("verifyClaim", () => {
     verifyClaim(validClaim, DISPENSING_USER_SCOPE, "3415870201", "test_sds_role_id")
     expect(console.warn).toHaveBeenCalled()
   })
+
+  test("raise an error if no endorsement code is provided in the claim", () => {
+    const result = verifyClaim(invalidClaim, DISPENSING_USER_SCOPE, "test_sds_user_id", "test_sds_role_id")
+    expect(result[0].severity).toEqual("error")
```
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply packages/coordinator/tests/services/validation/claim-validator.spec.ts]: The diagnostic message has been used in the test.

File: packages/coordinator/src/services/validation/claim-validator.ts (hunk 1 above)
Comment (kris-szlapa) [reply packages/coordinator/src/services/validation/claim-validator.ts]: The medicationDispenseEndorsementPresent function has been added.

File: packages/coordinator/src/services/validation/claim-validator.ts [hunk 3]
```diff
@@ -61,8 +61,12 @@ export function verifyClaim(
     return permissionErrors
   }
 
-  const medicationDispenseEndorsement = getMedicationDispenseEndorsement(claim)
-  if (!medicationDispenseEndorsement.length) {
+  function medicationDispenseEndorsementPresent(claim: fhir.Claim) {
```
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply packages/coordinator/src/services/validation/claim-validator.ts]: The code has been refactored.

**Review by originalphil — APPROVED** [review]

**Pull request merged**